src/App.js

```diff
@@ -1,28 +1,49 @@
 import { useEffect, useState } from "react"
-import { Link, Route, Routes } from "react-router-dom"
+import { Link, Route, Routes, useLocation } from "react-router-dom"
 import ContactsList from "./components/ContactsList"
 import ContactsAdd from "./components/ContactsAdd"
 import ContactsView from "./components/ContactsView"
+import ContactsEdit from "./components/ContactsEdit"
 import "./styles/styles.css"
 
 export default function App() {
   const [contacts, setContacts] = useState([])
-  
+  const [loading , setLoading] = useState(false)
+    const location = useLocation()
   //TODO: Load all contacts on useEffect when component first renders
+  useEffect(() => {
+    setLoading(true)
+    fetch("http://localhost:4000/contacts")
+    .then((response) => response.json())
+    
+    .then((data) => {
+      console.log("data from server",data)
+      setContacts(data)
+      setLoading(false)
+    })
+  }, [location])
```

Comment:
Using location here -  are you using the change of location as the dependency, so whenever the location changes you auto-reFetch the data?